address_book_ui/src/app/components/contact/contact.component.ts

```diff
@@ -0,0 +1,90 @@
+import { Component } from '@angular/core';
+import { CardModule } from 'primeng/card';
+import { ButtonModule } from 'primeng/button';
+import { BadgeModule } from 'primeng/badge';
+import { InputTextModule } from 'primeng/inputtext';
+import { FormsModule } from '@angular/forms';
+import { NgFor, NgIf, AsyncPipe } from '@angular/common';
+import { TableModule } from 'primeng/table';
+import { ChangeDetectorRef } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { ContactTableComponent } from '../contact-table/contact-table.component';
+import { ContactBlockComponent } from '../contact-block/contact-block.component';
+import { TagModule } from 'primeng/tag';
+
+@Component({
+  selector: 'app-contact',
+  standalone: true,
+  imports: [CardModule,
+    ButtonModule,
+    BadgeModule,
+    InputTextModule,
+    FormsModule,
+    NgFor,
+    NgIf,
+    AsyncPipe,
+    TableModule,
+    ContactTableComponent,
+    ContactBlockComponent,
+    TagModule],
+  templateUrl: './contact.component.html',
+  styleUrl: './contact.component.scss'
+})
+export class ContactComponent {
+  constructor(private cdr: ChangeDetectorRef) { }
+
+  name: string = "First Name";
+  subheader: string = "+12345678";
+  streetAddress: string = '';  // To store street/precise address
+  location: string = '';  // To store location
+
+  contactDetails = [
+    {
+      type: 'Email',
+      value: 'john.doe@example.com',
+      isDisabled$: new BehaviorSubject<boolean>(true),
+      placeholders: ["gmail.com"]
```

Comment:
* because some strings are repeated here, you can create a constants.ts file to export common strings. This way, any future updates to the string values can be made by only modifying the constant

Reply:
This is just a placeholder, so that I have an idea how it will look like